### DISCUSSION TITLE
Error with .set(name) after destroying Tabview

### DISCUSSION THREAD

**Discussion by PrizeGotti** [TomSchimansky/CustomTkinter#1214] [Q&A]
In my application, I need to destroy the tabview to reload it with updated labels.

This all works fine except when I try to set the visible tab back to what it was before I destroyed the tabview.

I get the following error:

> _tkinter.TclError: bad window path name ".!ctkframe2.!ctktabview.!ctkframe"

How can I fix this issue so I can set the visible tab back to the correct one?

Many thanks!

**Comment by siddheshgunjal** [comment]
Please share some sample code to reproduce the error. Then the community can help

**Reply by PrizeGotti** [reply]
My apologies, didn't see your reply as it went to my work email.

```
import customtkinter as tk
import pandas as pd
import os, time
from PIL import Image, ImageTk
from customtkinter import CTkImage
from tkinter import filedialog as fd

# Create main window, geometry, icon and fonts
root = tk.CTk()
root.geometry("1280x720")
root.title("Nene Document Control")
root.iconbitmap("NeneDoc/Meadow.ico")
tk.set_appearance_mode("dark")
headerfont = tk.CTkFont(family="Quatro UltraBlack", size=32, weight="bold")
mainfont = tk.CTkFont(family="Calibri", size=14, weight="bold")

# Create header logo and title
header = tk.CTkFrame(root)
header.pack(side="top" , fill="both")
MeadowLogo = CTkImage(dark_image=Image.open(r"NeneDoc/Meadow.png"), size=(100 , 100))
LogoHolder = tk.CTkLabel(master=header, image=MeadowLogo, text="").grid(column=0, row=0)
headertitle = tk.CTkLabel(master=header, text="Nene Document Control", font=headerfont).grid(column=1, row=0)

# Create main frame
main = tk.CTkFrame(root)
main.pack(side="top" , fill="both", expand=True)

# Create frames, tabs and label
def create_frames():
    # Read data from the "NeneDoc.xlsx" file
    xlsmap = pd.read_excel("NeneDoc/NeneDoc.xlsx", sheet_name=None)
    tabview = tk.CTkTabview(main)
    tabview.pack(padx=5, pady=5, fill="both", expand=True)

    for i in xlsmap:
        
        data = pd.read_excel("NeneDoc/NeneDoc.xlsx", sheet_name=i)
        tabname = str(i)
        tabview.add(tabname)

        # Create "Add Document" button
        topbar = tk.CTkFrame(tabview.tab(tabname), fg_color="#333333")
        topbar.pack(padx=21, pady=5, fill="x")
        addnew = tk.CTkButton(topbar, text="Add Document", command=lambda: add_doc(), fg_color="Green", hover_color="Dark Green", font=mainfont, width=155)
        addnew.pack(anchor="ne", padx=5)

        # Create scrollable frame for tabs
        tabframe = tk.CTkScrollableFrame(tabview.tab(tabname))
        tabframe.pack(padx=5, pady=5, fill="both", expand=True)

        # Create a container frame for each entry
        for i in range(len(data)):
            container_frame = tk.CTkFrame(tabframe, height=35)
            container_frame.pack(padx=5, pady=0, fill="x", expand=False)
            container_frame.pack_propagate(0)
            
            # Create the first frame
            frame1 = tk.CTkFrame(container_frame, width=320)
            frame1.pack(padx=5, pady=5, side="left", fill="x", expand=False)
            frame1.pack_propagate(0)
            tk.CTkLabel(frame1, text=data["FileName"][i], font=mainfont).pack()
            
            # Create the second frame
            frame2 = tk.CTkFrame(container_frame, width=710)
            frame2.pack(padx=5, pady=5, side="left", fill="x", expand=True)
            frame2.pack_propagate(0)
            tk.CTkLabel(frame2, text=data["FileLocation"][i], font=mainfont, anchor="w").pack(side="left", padx=10)
            
            # Create the third frame
            frame3 = tk.CTkFrame(container_frame, width=120, fg_color="#2b2b2b")
            frame3.pack(padx=5, pady=5, side="left", fill="x", expand=False)
            frame3.pack_propagate(0)
            tk.CTkButton(frame3, text="Open", command=lambda file_location=data["FileLocation"][i]: open_doc(file_location), font=mainfont).pack()

            # Create the forth frame
            frame4 = tk.CTkFrame(container_frame, width=30, fg_color="#2b2b2b")
            frame4.pack(padx=0, pady=5, side="right", fill="x", expand=False)
            frame4.pack_propagate(0)
            tk.CTkButton(frame4, fg_color="Red", hover_color="Dark Red", text="X", command=lambda file_location=data["FileLocation"][i]: del_entry(file_location), font=mainfont, width=30).pack()

        # Create function to open linked document
        def open_doc(file_location):
            filetoopen = file_location.replace("\\", "\\\\")
            os.startfile(filetoopen)

        # Create function to add document to list
        def add_doc():
            tname = tabview.get()
            dcount = pd.read_excel("NeneDoc/NeneDoc.xlsx", sheet_name=tname).shape[0]
            
            entname = tk.CTkInputDialog(text="Enter Document Name", title="Document Name").get_input()
            entloc = fd.askopenfilename(title="Select Document To Add", initialdir='/')

            entdata = {"FileName": [entname], "FileLocation": [entloc]}
            entdf = pd.DataFrame(data=entdata)

            writer = pd.ExcelWriter("NeneDoc/NeneDoc.xlsx", mode="a", engine="openpyxl", if_sheet_exists="overlay")
            entdf.to_excel(writer, sheet_name=tname, header=False, index=False, startrow=dcount+1)
            writer.close()

            tabview.destroy()
            create_frames()
            tabview.set(tname)

        # Create function to delete document from list
        def del_entry(file_location):
            tname = tabview.get()
            deldata = pd.read_excel("NeneDoc/NeneDoc.xlsx", sheet_name=tname)

            deldata = deldata.loc[deldata["FileLocation"] != file_location]

            writer = pd.ExcelWriter("NeneDoc/NeneDoc.xlsx", mode="a", engine="openpyxl", if_sheet_exists="replace")
            deldata.to_excel(writer, sheet_name=tname, index=False)
            writer.close()

            tabview.destroy()
            create_frames()
            tabview.set(tname)

create_frames()

root.pack_propagate(0)

root.mainloop()
```

When I run either the "add_doc" or "del_entry" functions, I get the following errow when I call "tabview.set(tname)" in either function.

> Exception in Tkinter callback
Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.2800.0_x64__qbz5n2kfra8p0\lib\tkinter\__init__.py", line 1921, in __call__
    return self.func(*args)
  File "C:\Users\hallc\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\customtkinter\windows\widgets\ctk_button.py", line 553, in _clicked
    self._command()
  File "c:\Users\hallc\Python Projects\NeneDoc\NeneDoc CTk.py", line 79, in <lambda>
    tk.CTkButton(frame4, fg_color="Red", hover_color="Dark Red", text="X", command=lambda file_location=data["FileLocation"][i]: del_entry(file_location), font=mainfont, width=30).pack()
  File "c:\Users\hallc\Python Projects\NeneDoc\NeneDoc CTk.py", line 118, in del_entry
    tabview.set(tname)
  File "C:\Users\hallc\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\customtkinter\windows\widgets\ctk_tabview.py", line 363, in set
    self._grid_forget_all_tabs()
  File "C:\Users\hallc\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\customtkinter\windows\widgets\ctk_tabview.py", line 176, in _grid_forget_all_tabs
    frame.grid_forget()
  File "C:\Users\hallc\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\customtkinter\windows\widgets\core_widget_classes\ctk_base_class.py", line 326, in grid_forget
    return super().grid_forget()
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.2800.0_x64__qbz5n2kfra8p0\lib\tkinter\__init__.py", line 2532, in grid_forget
    self.tk.call('grid', 'forget', self._w)
_tkinter.TclError: bad window path name ".!ctkframe2.!ctktabview.!ctkframe"

Many thanks!

**Reply by Wolf-SO** [reply]
First: thanks for sharing code to reproduce the issue. 😄 

Second: If this can not be shortened (really?), then at least fix the code boxes for better readability, this might be helpful [Creating and highlighting code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks "Creating and highlighting code blocks - GitHub Docs"). Thanks.

**Reply by PrizeGotti** [reply]
My apologies, I'm new to github and trying to get a grip with its formatting (overly complexed in comparison to other sites).

Hopefully that's easier to read. I thought it was best to provide the whole code so you can see what steps are taking place.

**Reply by Wolf-SO** [reply]
No need for apologies, we all are just learning. It's easier to read, definitively. I'll have a deeper look on this soon…

**Comment by Wolf-SO** [comment]
The example you gave is too complex. It needs several files to work: Symbol image, Excel file, font, etc. Please try to reduce it step by step, but keeping the error.

**Comment by whyming1105** [comment]
When I use AntoDock to dock molecules, this error occurs when generating a. blg file. How can I solve this problem?

Traceback (most recent call last):
  File "F:\autodock\mgltools-1.5.6\lib\site-packages\ViewerFramework\VF.py", line 898, in tryto
    result = command( *args, **kw )
  File "F:\autodock\mgltools-1.5.6\lib\site-packages\AutoDockTools\autostartCommands.py", line 968, in doit
    self.vf.ADstart_manage.addProcess(ps)
  File "F:\autodock\mgltools-1.5.6\lib\site-packages\AutoDockTools\autostartCommands.py", line 269, in addProcess
    if not self.kill.master.winfo_ismapped() and not self.kill.done:
  File "F:\autodock\mgltools-1.5.6\lib\lib-tk\Tkinter.py", line 749, in winfo_ismapped
    self.tk.call('winfo', 'ismapped', self._w))
TclError: bad window path name ".219297792"
